src/lib/rng/esdm_rng/esdm_rng.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+* ESDM RNG
+* (C) 2024, Markus Theil <theil.markus@gmail.com>
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <botan/esdm_rng.h>
+
+#include <esdm/esdm_rpc_client.h>
+
+namespace Botan {
+
+namespace {
+class ESDM_Context {
```

Comment:
```suggestion
/**
* This helper makes sure that the ESDM service is initialized and 
* finalized as needed in a threadsafe fashion. Finalization happens
* as soon as all instances of ESDM_RNG are destructed. This may
* happen multiple times in the lifetime of the process.
*/
class ESDM_Context {
```

... future-us will be grateful. 😄